aptos-move/framework/aptos-framework/sources/transaction_validation.move

```diff
@@ -290,43 +315,53 @@ module aptos_framework::transaction_validation {
+                    error::out_of_range(PROLOGUE_ECANT_PAY_GAS_DEPOSIT),
+                );
+            };
 
-        let amount_to_burn = if (features::collect_and_distribute_gas_fees()) {
-            // TODO(gas): We might want to distinguish the refundable part of the charge and burn it or track
-            // it separately, so that we don't increase the total supply by refunding.
+            let amount_to_burn = if (features::collect_and_distribute_gas_fees()) {
+                // TODO(gas): We might want to distinguish the refundable part of the charge and burn it or track
+                // it separately, so that we don't increase the total supply by refunding.
 
-            // If transaction fees are redistributed to validators, collect them here for
-            // later redistribution.
-            transaction_fee::collect_fee(gas_payer, transaction_fee_amount);
-            0
-        } else {
-            // Otherwise, just burn the fee.
-            // TODO: this branch should be removed completely when transaction fee collection
-            // is tested and is fully proven to work well.
-            transaction_fee_amount
-        };
+                // If transaction fees are redistributed to validators, collect them here for
+                // later redistribution.
+                transaction_fee::collect_fee(gas_payer, transaction_fee_amount);
+                0
+            } else {
+                // Otherwise, just burn the fee.
+                // TODO: this branch should be removed completely when transaction fee collection
+                // is tested and is fully proven to work well.
+                transaction_fee_amount
+            };
 
-        if (amount_to_burn > storage_fee_refunded) {
-            let burn_amount = amount_to_burn - storage_fee_refunded;
-            transaction_fee::burn_fee(gas_payer, burn_amount);
-        } else if (amount_to_burn < storage_fee_refunded) {
-            let mint_amount = storage_fee_refunded - amount_to_burn;
-            transaction_fee::mint_and_refund(gas_payer, mint_amount)
+            if (amount_to_burn > storage_fee_refunded) {
+                let burn_amount = amount_to_burn - storage_fee_refunded;
+                transaction_fee::burn_fee(gas_payer, burn_amount);
+            } else if (amount_to_burn < storage_fee_refunded) {
+                let mint_amount = storage_fee_refunded - amount_to_burn;
+                transaction_fee::mint_and_refund(gas_payer, mint_amount)
+            };
         };
 
         // Increment sequence number
         let addr = signer::address_of(&account);
         account::increment_sequence_number(addr);
     }
+
+    inline fun skip_auth_key_check(is_simulation: bool, auth_key: &vector<u8>): bool {
+        is_simulation && vector::is_empty(auth_key)
+    }
+
+    inline fun skip_gas_payment(is_simulation: bool, gas_payer: address): bool {
+        is_simulation && gas_payer == @0x0
```

Comment:
So, I might have missed it, but there's no way for someone to override the is_simulation at all in a real VM right?

Reply:
`is_simulation` can be true only inside of `AptosSimulationVM` (`pub struct AptosSimulationVM(AptosVM);`).
https://github.com/aptos-labs/aptos-core/blob/07e9f7866b88857f0ca1d7ea6f49529575bf9617/aptos-move/aptos-vm/src/aptos_vm.rs#L2721C12-L2721C32